Add type definitions for Lua bitop

matt-allan opened the pull request:
These aren't automatically included unless you have the runtime set to Lua JIT, which we don't want for renoise tools. Unfortunately there isn't an existing addon for lua bitop I could find to include. Inlining it seems to be the norm in other addons from the LLS addons repo.

emuell commented:
This will probably be obsolete in the future as we plan to switch to LuaJIT, but for now it's missing, so it's very useful. Thanks!

emuell reviewed (commented):
Could you please change the header to match the rest of the files? Then I'll be perfectly happy with the formatting :)

```lua
---@meta
error("Do not try to execute this file. It's just a type definition file.")
---
---Please read the introduction at https://github.com/renoise/xrnx/
---to get an overview about the complete API, and scripting for
---Renoise in general...
---

-------------------------------------------------------------------------------

---Integer, Bit Operations, provided by <http://bitop.luajit.org/>
---
--- [Documentation](http://bitop.luajit.org/api.html)
bit = {}

---Normalizes a number to the numeric range for bit operations and returns it.
...
```

Also `bit = {}` seems to be needed to silence some warnings from LuaLS.

matt-allan commented:
Thanks for the quick review!

I didn't mean to leave `bit = {}` out; that was a mistake when I moved it from `renoise.lua`. Fixed in the latest commit.

emuell commented:
Thanks for the contribution. Looks all good now.
